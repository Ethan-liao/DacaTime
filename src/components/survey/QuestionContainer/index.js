import React from 'react';
import Icon from '../../basic/Icon';
import Panel from '../../basic/Panel';
import Modal from '../../basic/Modal';

class QuestionContainer extends React.Component {
  constructor(props) {
    super();

    this.state = {
      modalIsOpen: false,
    };

    this.openModal = this.openModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
  }

  render() {
    const explanationLink = !this.props.explanation
      ? null
      : <div className="dt-question-explanation">
          <a onClick={this.openModal}>What does this mean?</a>
        </div>;

    const modal = !this.props.explanation
      ? null
      : <div>
          <Modal open={this.state.modalIsOpen} onClose={this.closeModal}>
            <Icon name="fa-info-circle" />&nbsp;
            {this.props.explanation}
            {this.state.modalIsOpen}
          </Modal>
        </div>;

    return (
      <Panel className="dt-question-container">
        <div className="dt-question-header">
          {this.props.prompt}
        </div>
        {explanationLink}
        {this.props.children}
        {modal}
      </Panel>
    );
  }

  closeModal() {
    this.setState({
      modalIsOpen: false,
    });
  }

  openModal() {
    this.setState({
      modalIsOpen: true,
    });
  }
}

export default QuestionContainer;
