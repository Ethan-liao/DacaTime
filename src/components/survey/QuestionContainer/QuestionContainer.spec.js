import React from 'react';
import QuestionContainer from './index';
import renderer from 'react-test-renderer';
import { questionWithExplanation } from '../../../util/testData';
import reducer from '../../../reducers';
import { setup } from '../../../util/testUtils';

const { component, store } = setup(
  QuestionContainer,
  reducer,
  questionWithExplanation,
);

it('renders without crashing', () => {
  renderer.create(component);
});

it('renders correctly', () => {
  const renderedComponent = renderer.create(component);
  let tree = renderedComponent.toJSON();

  expect(tree).toMatchSnapshot();
});
