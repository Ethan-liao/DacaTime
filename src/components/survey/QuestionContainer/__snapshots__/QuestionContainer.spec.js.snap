exports[`test renders correctly 1`] = `
<div
  className="dt-panel dt-question-container">
  <div
    className="dt-question-header">
    Have you been convicted of a felony?
  </div>
  <div
    className="dt-question-explanation">
    <a
      onClick={[Function]}>
      What does this mean?
    </a>
  </div>
  <div />
</div>
`;
